packages/in_app_purchase/in_app_purchase_android/lib/src/types/google_play_purchase_details.dart

```diff
@@ -26,7 +26,7 @@ class GooglePlayPurchaseDetails extends PurchaseDetails {
   factory GooglePlayPurchaseDetails.fromPurchase(PurchaseWrapper purchase) {
     final GooglePlayPurchaseDetails purchaseDetails = GooglePlayPurchaseDetails(
       purchaseID: purchase.orderId,
-      productID: purchase.sku,
+      productID: purchase.products.isNotEmpty ? purchase.products.first : '',
```

Comment:
Could you add a comment here explaining why it's okay to drop the other products?

Reply:
Sorry to interrupt. I have gone through [this issue](https://github.com/flutter/flutter/issues/110909#issuecomment-1501723484)  and have a simple question!

In my opinion, to use a single class that covers both Google Play Store and App Store like `PurchaseDetails` and `ProductDetails`, we have to pretend 1 Subscription must have only 1 Base Plan in Google Play Store.

**Other base plans will be dropped if we only handle the first base plan*

I wondered if dropping other base plans in Google Play Store and noticing in package usage docs or sth is okay.

here is [the link](https://github.com/flutter/flutter/issues/110909#issuecomment-1408238392) I previously added comment in issue. thanks

Reply:
Hi @seunghwanly 👋 

The approach this PR takes is returning all base plans and offers instead of only a single one, as is requested in [110909](https://github.com/flutter/flutter/issues/110909). The logic for splitting out the original object into multiple is here:
https://github.com/flutter/packages/blob/9877ef6b4a77cce4e227c104ae43e8c807f0e374/packages/in_app_purchase/in_app_purchase_android/lib/src/types/google_play_product_details.dart#L93-L123

Reply:
Thanks for the reply @JeroenWeener  👍  

I really appreciate your consideration.